docs/rest_api_guide/05_RESTdeveloper.rst

```diff
@@ -0,0 +1,132 @@
+
+  - *service_name* - the name of the service in which the plugin is running
+
+  - *plugin_name* - the name of the plugin within the service. For a north or south plugin this is the same as the service name. For a filter this will be the name given to the filter instance when it was added to the pipeline.
+
+**Response Payload**
+
+The response payload is the persisted data from the plugin.
+
+**Example**
+
+.. code-block:: console
+
+   $ curl http://localhost:8081/fledge/service/OMF/plugin/OMF/data
+
+Where *OMF* is the name of a north service with an OMF filter connected to a PI Server. In this case the persisted data is the type information we cache locally that describes the types that have been created within the OMF layer of the PI Server.
+
+.. code-block:: console
+
+    {
+      "data": {
+        "sentDataTypes": [
+          {
+            "sine25": {
+              "type-id": 1,
+              "dataTypesShort": "0x101",
+              "hintChecksum": "0x0",
+              "namingScheme": 0,
+              "afhHash": "15489826335467873671",
+              "afHierarchy": "fledge/data_piwebapi/mark",
+              "afHierarchyOrig": "fledge/data_piwebapi/mark",
+              "dataTypes": {
+                "sinusoid": {
+                  "type": "number",
+                  "format": "float64"
+                }
+              }
+            }
+          },
+          {
+            "sinusoid": {
+              "type-id": 1,
+              "dataTypesShort": "0x101",
+              "hintChecksum": "0x0",
+              "namingScheme": 0,
+              "afhHash": "15489826335467873671",
+              "afHierarchy": "fledge/data_piwebapi/mark",
+              "afHierarchyOrig": "fledge/data_piwebapi/mark",
+              "dataTypes": {
+                "sinusoid": {
+                  "type": "number",
+                  "format": "float64"
+                }
+              }
+            }
+          }
+        ]
+      }
+    }
+
+.. note::
+
+   Persisted data is written when the plugin is shutdown. Therefore in order to obtain accurate results this call should only be made when the service is shutdown. Calling this API when a service is running will result in data from the previous time the service was shutdown.
+
+``POST /fledge/service/{service_name}/plugin/{plugin_name}/data`` - write the persisted data for a plugin
+
+Write or overwrite data persisted by the plugin. The request payload is the data which the plugin should receive and must be in the correct format for the plugin.
+
+.. note::
+
+   Persisted data is written when the plugin is shutdown. Therefore in order to obtain predictable results this call should only be made when the service is shutdown. Calling this API when a service is running will result in data that is written by the call being overwritten by the plugin when it shuts down.
+
+
+``DELETE /fledge/service/{service_name}/plugin/{plugin_name}/data`` - delete the persisted data for the plugin
```

Comment:
Can we add delete only if service is not in running state?

Reply:
That's already there as the note shown below it.